### DISCUSSION TITLE
Can clone and push from one computer but not another. Same account

### DISCUSSION THREAD

**Discussion by Denis-Kazakov** [community/community#148073] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

I have personal repos and I am also a member of an organization, but have different experiences with my organization's repos depending on the computer I use, though account is the same.

In fact, I tried four computers: two with Windows and two with Linux Mint.

On both Windows machines, I can create repos, clone them, push to them with command line or GUI tool, without any authentication hassle. No requests for tokens or anything.

On Linux machines, I can log in to my GitHub account, visit all public and private repose, view any files in the organization's repos but can only clone and push to my personal repos. I created a token with access to all repositories and write and read permission for Contents, but it does not help. What I get is:
`remote: Write access to repository not granted.
fatal: unable to access ...: The requested URL returned error: 403`

I was able to push my PyCharm projects to the organizations repos somehow. I think with browser authentication.

**Comment by lemorage** [comment, marked as answer]
Try to use SSH Auth on linux machines.

1. Generate an SSH key on the Linux machine:
```sh
ssh-keygen -t rsa -b 4096 -C "your_email@example.com"
```
  - Save it to the default location (~/.ssh/id_rsa) or a custom path.
<br/>

2. Add the public key to GitHub:
  - Copy the public key:
```sh
cat ~/.ssh/id_rsa.pub
```
  - Go to GitHub > Settings > SSH and GPG keys > New SSH key.
  - Paste the public key and save.
<br/>

3. Update your Git remote URL to use SSH:
```sh
git remote set-url origin git@github.com:organization/repo.git
```
<br/>

4. Test the SSH connection:
```sh
ssh -T git@github.com
```
You should see a message confirming your access.

---

Since pushing via PyCharm is possible, maybe you could also try to use the same mechanism on your linux machines:
1. Check PyCharm’s Git settings:
  - Go to Settings > Version Control > Git.
  - Check if it’s using a token or SSH for authentication.
2. Replicate this configuration in your terminal Git setup:
  - If PyCharm uses an SSH key, ensure the key is added to your SSH agent:
```sh
eval "$(ssh-agent -s)"
ssh-add ~/.ssh/id_rsa
```

**Reply by Denis-Kazakov** [reply]
Thank you! It worked. 
I am still puzzled by cloning with HTTPS but at least can do it with SSH.
